gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java

```diff
@@ -0,0 +1,737 @@
+                            "Not able to identify the plan to update: " +
+                            planNode.get("name").asText() +
+                            ". Too much plan with the same name"
+                        );
+                    }
+                }
+            }
+        }
+        // Metadata
+        final JsonNode metadataDefinition = jsonNode.path("metadata");
+        if (metadataDefinition != null && metadataDefinition.isArray()) {
+            try {
+                for (JsonNode metadataNode : metadataDefinition) {
+                    UpdateApiMetadataEntity updateApiMetadataEntity = objectMapper.readValue(
+                        metadataNode.toString(),
+                        UpdateApiMetadataEntity.class
+                    );
+                    updateApiMetadataEntity.setApiId(createdOrUpdatedApiEntity.getId());
+                    apiMetadataService.update(updateApiMetadataEntity);
+                }
+            } catch (Exception ex) {
+                LOGGER.error("An error occurs while creating API metadata", ex);
+                throw new TechnicalManagementException("An error occurs while creating API Metadata", ex);
+            }
+        }
+    }
+
+    private static class MemberToImport {
+
+        private String source;
+
+        private String sourceId;
+        private List<String> roles; // After v3
+        private String role; // Before v3
+
+        public MemberToImport() {}
+
+        public MemberToImport(String source, String sourceId, List<String> roles, String role) {
+            this.source = source;
+            this.sourceId = sourceId;
+            this.roles = roles;
+            this.role = role;
+        }
+
+        public String getSource() {
+            return source;
+        }
+
+        public void setSource(String source) {
+            this.source = source;
+        }
+
+        public String getSourceId() {
+            return sourceId;
+        }
+
+        public void setSourceId(String sourceId) {
+            this.sourceId = sourceId;
+        }
+
+        public List<String> getRoles() {
+            return roles;
+        }
+
+        public void setRoles(List<String> roles) {
+            this.roles = roles;
+        }
+
+        public String getRole() {
+            return role;
+        }
+
+        public void setRole(String role) {
+            this.role = role;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            MemberToImport that = (MemberToImport) o;
+
+            return source.equals(that.source) && sourceId.equals(that.sourceId);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = source.hashCode();
+            result = 31 * result + sourceId.hashCode();
+            return result;
+        }
+    }
+
+    private static class PageEntityTreeNode {
```

Comment:
same ?

Reply:
Discussed, we will do it in another PR because it needs some extra refactoring.